xterm/index.cgi

```diff
@@ -52,26 +111,32 @@ $ENV{'SESSION_ID'} = $main::session_id;
 sleep(1);
 
 # Open the terminal
-print "<center>\n";
-print &ui_table_start(undef, undef, 2);
-print &ui_table_row(undef, "<div id=terminal></div>", 2);
-print &ui_table_end();
-print "</center>\n";
 my $url = "wss://".$ENV{'HTTP_HOST'}.$wspath;
-my $rows = $config{'rows'} || 24;
-my $cols = $config{'cols'} || 80;
+print "<div id=\"terminal\" $termopts->{'ContainerStyle'}></div>\n";
+my $term_script = <<EOF;
+	var term = new Terminal($termopts->{'Options'}),
+		socket = new WebSocket('$url', 'binary'),
+		attachAddon = new AttachAddon.AttachAddon(socket);
+	term.loadAddon(attachAddon);
+	$termopts->{'FitAddonLoad'}
+	term.open(document.getElementById('terminal'));
+	$termopts->{'FitAddonAdjust'}
+	term.focus();
+EOF
+
+# Return inline script data depending on type
+my $term_script_data =
```

Comment:
What does this do?

Reply:
Themes that are designed as SPA won't be able to process module load in a standard way, as there is no regular page load process happening. The module instead will return JSON object with instructions from the server.

Reply:
Surely perlcritic would complain about it being empty.

Reply:
Oh, I see it continues on the next line. Sorry. I'm just confused.

Reply:
What's special about this module that the theme needs to get a JSON object?

Reply:
This module preloads `xterm` _js_ and _css_. Authentic theme when run in a SPA mode (default, with the nav menu) doesn't load things from the `head` like this. Also, as was in development it's easier for me like this, as I will have a simpler and more deep control over the module on the theme side.

Reply:
I see, that makes sense